api/src/main/java/com/graphhopper/internal/LazyPointList.java

```diff
@@ -0,0 +1,152 @@
+package com.graphhopper.internal;
+
+
+import com.graphhopper.bean.RoutePoint;
+
+import java.util.*;
+
+public class LazyPointList implements List<RoutePoint> {
```

Comment:
Yes, http://it.wikipedia.org/wiki/Lazy_initialization. Is a performance optimization, hidden to the end-user (internal). "Lazy" is the global and universal name for it, in all languages ever.